File: java/StandardDeviation.java
package formucode;

public class StandardDeviation {

	protected double standardDeviation(final double[] someData){

	    double sum = 0.0;
	    double mean = 0.0;
	    double n = 0;

	    for(double data : someData){
	        sum += data;
	        n++;
	    }

	    mean = sum / n;

	    sum = 0;

	    for(double data : someData){
	        sum += Math.pow((data - mean), 2);
	    }
	    //using e ^ (natural log / root) improves accuracy
	    double stdDev = Math.pow(Math.E, (Math.log(sum/n)/2));

	    return stdDev;
	}
}

File: javascript/standardDeviation.js

standardDeviation = (numArr) => {

    let sum = 0;
    let n = 0;

    numArr.forEach((val)=>{
        sum += val;
        n ++;
    })

    const mean = sum / n;
    let summation = 0;

    numArr.forEach((val)=>{
        summation += Math.pow((val-mean) , 2);
    })

    const stdDev = Math.pow((summation/n), (1/2));

    return stdDev;
}